lib/src/rules/lines_longer_than_80_chars.dart

```diff
@@ -0,0 +1,191 @@
+      rule.reporter
+          .reportErrorForOffset(rule.lintCode, line.offset, line.length);
+    }
+  }
+}
+
+class _LineInfo {
+  _LineInfo({this.index, this.offset, this.end});
+  final int index;
+  final int offset;
+  final int end;
+  int get length => end - offset;
+}
+
+class _AllowedLongLineVisitor extends RecursiveAstVisitor {
+  _AllowedLongLineVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitStringInterpolation(StringInterpolation node) {
+    if (node.isMultiline) {
+      _handleMultilines(node);
+    } else {
+      final value = node.elements.map((e) {
+        if (e is InterpolationString) return e.value;
+        if (e is InterpolationExpression) return ' ' * e.length;
+      }).join();
+      _handleSingleLine(node, value);
+    }
+  }
+
+  @override
+  visitSimpleStringLiteral(SimpleStringLiteral node) {
+    if (node.isMultiline)
+      _handleMultilines(node);
+    else
+      _handleSingleLine(node, node.value);
+  }
+
+  _handleMultilines(SingleStringLiteral node) {
+    final startLine = lineInfo.getLocation(node.offset).lineNumber;
+    final endLine = lineInfo.getLocation(node.end).lineNumber;
+    for (var i = startLine; i <= endLine; i++) {
+      allowedLines.add(i);
+    }
+  }
+
+  _handleSingleLine(AstNode node, String value) {
+    if (_looksLikeUriOrPath(value)) {
+      final line = lineInfo.getLocation(node.offset).lineNumber;
+      allowedLines.add(line);
+    }
+  }
+}
+
+class _AllowedCommentVisitor extends SimpleAstVisitor {
+  _AllowedCommentVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitCompilationUnit(CompilationUnit node) {
+    var token = node.beginToken;
+    while (token != null) {
+      _getPrecedingComments(token).forEach(_visitComment);
+      if (token == token.next) break;
+      token = token.next;
+    }
+  }
+
+  Iterable<CommentToken> _getPrecedingComments(Token token) sync* {
+    CommentToken comment = token.precedingComments;
+    while (comment != null) {
+      yield comment;
+      comment = comment.next;
+    }
+  }
+
+  void _visitComment(CommentToken comment) {
+    final content = comment.toString();
+    final lines = [];
+    if (content.startsWith('///')) {
+      lines.add(content.substring(3));
+    } else if (content.startsWith('//')) {
+      lines.add(content.substring(2));
+    } else if (content.startsWith('/*')) {
+      // remove last slash before finding slash
+      lines.addAll(content.substring(2, content.length - 2).split('\n'));
+    }
+    for (var i = 0; i < lines.length; i++) {
+      final value = lines[i];
+      if (_looksLikeUriOrPath(value)) {
+        final line = lineInfo.getLocation(comment.offset).lineNumber + i;
+        allowedLines.add(line);
+      }
+    }
+  }
+}
+
+/// String looks like URI if it contains a slash or backslash followed by 5
+/// non-empty chars.
```

Comment:
I'm not sure where the 5 came from. More importantly, it seems like this pattern will match some strings that it shouldn't match ('sin = opposite/hypotenuse') and not match some strings that it ought to match ('C:\a directoryPath\with spacesInTheComponents'). Are there any false positives when running this over real code?

Reply:
I changed the rule to check only the presence of a `/` or `\`. It's quite impossible to tell if a String contains an URI or path because almost everything can be an URI. So I think it's better to have false negative (because a String or comment contains a `/` or `\`) rather than false positive that will force to split a real URI on two lines to remove the lint.